server/src/main/java/org/elasticsearch/action/support/broadcast/node/BoundedDiagnosticRequestPermits.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.action.support.broadcast.node;
+
+import org.elasticsearch.common.settings.ClusterSettings;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.util.concurrent.AdjustableSemaphore;
+
+/**
+ * This class guards the amount of bounded diagnostic requests a node can concurrently coordinate.
+ */
+public class BoundedDiagnosticRequestPermits {
+
+    public static final Setting<Integer> MAX_CONCURRENT_BOUNDED_DIAGNOSTIC_REQUESTS_PER_NODE = Setting.intSetting(
+        "cluster.max_concurrent_bounded_diagnostic_requests_per_node",
+        20,
+        1,
+        Setting.Property.NodeScope,
+        Setting.Property.Dynamic
+    );
+
+    private final AdjustableSemaphore maxConcurrentBoundedDiagnosticRequestsPerNodeSemaphore;
+    private volatile Integer maxConcurrentBoundedDiagnosticRequestsPerNode;
```

Comment:
It doesn't appear that we use this integer anywhere, outside of when we're creating or modifying the semaphore. Can we remove it since the integer value is unneeded?

Reply:
You are right, I used it during my development to double check that it worked as expected, but it doesn't add anything in the long run.... I will remove it.